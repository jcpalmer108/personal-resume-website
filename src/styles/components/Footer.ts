import styled from "styled-components";
import { desktopBreakpoint, redstone, silver_polish, tabletBreakpoint } from "../variables";

export const Wrapper = styled.div`
  padding: 40px;
  text-align: center;

  @media (min-width: ${desktopBreakpoint}px) {
    padding: 70px;
  }
`

export const Logo = styled.img`
  height: 55px;
  padding-bottom: 40px;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 75px;
    padding-bottom: 52px;
  }

  @media (min-width: ${tabletBreakpoint}px) {
    height: 90px;
    padding-bottom: 35px;
  }
`

export const FooterMenu = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  max-width: 669px;
  margin: auto;
  height: 218px;
  font-size: 14px;
  font-family: Gilroy-Medium;

  @media (min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    height: 14px;
  }

`

export const FooterPages = styled.a`
  text-decoration: none;
  color: ${silver_polish};

  &:hover {
    color: ${redstone};
  }

`

export const Copyright = styled.p`
  color: ${silver_polish};
  text-align: center;
`

export const Name = styled.span`
  color: ${redstone};
`