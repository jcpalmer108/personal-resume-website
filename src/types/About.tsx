import { Sections } from "./content";

export type AboutProps = {
  content: Sections | undefined;
};
