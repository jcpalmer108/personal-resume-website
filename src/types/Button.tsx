export type ButtonProps = {
  label: string;
  url?: string;
  onClick?: Function;
};
