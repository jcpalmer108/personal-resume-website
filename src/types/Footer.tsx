import { MenuItem } from "./global";

export type FooterProps = {
  menu: MenuItem[];
};
