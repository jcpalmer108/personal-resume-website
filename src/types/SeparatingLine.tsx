export type SeparatingLineProps = {
  thin?: boolean;
  light?: boolean;
  always?: boolean;
};
