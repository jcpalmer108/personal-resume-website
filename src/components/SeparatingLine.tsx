import { Line } from "../styles/components/SeparatingLine";
import { SeparatingLineProps } from "../types/SeparatingLine";

export default function SeparatingLine({
  light,
  thin,
  always,
}: SeparatingLineProps) {
  return <Line light={light} thin={thin} always={always} data-testid="Line" />;
}
