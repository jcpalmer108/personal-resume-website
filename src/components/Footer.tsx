import { 
  Logo, 
  Wrapper, 
  FooterMenu, 
  FooterPages, 
  Copyright, 
  Name
} from '../styles/components/Footer';
import Section from './Section';
import logo from '../assets/images/logos/jennapalmer-light.svg';
import { FooterProps } from '../types/Footer';
import { MenuItem } from '../types/global';
import SeparatingLine from './SeparatingLine';

export default function Footer({ menu }: FooterProps) {
  return (
    <>
      <Section thin wide >
        <Wrapper data-testid="Footer">
          <Logo src={logo}/>
          <FooterMenu data-testid="Footer Menu">
            {menu.map((link: MenuItem) => (
              <FooterPages href={`#${link.key}`} key={link.key}>
                {link.label?.toUpperCase()}
              </FooterPages>
            ))}
          </FooterMenu>
        </Wrapper>
      </Section>
      <SeparatingLine thin always />
      <Section thin>
        <Copyright data-testid="Copyright">COPYRIGHT &copy; ALL RIGHTS RESERVED BY <Name>JENNA PALMER</Name></Copyright>
      </Section>    
    </>
  )
}