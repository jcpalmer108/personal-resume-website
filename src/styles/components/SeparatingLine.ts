import styled from "styled-components";
import { desktopBreakpoint, silver_polish } from "../variables";
import { SeparatingLineProps as LineProps } from "../../types/SeparatingLine";

export const Line = styled.hr<LineProps>`
  width: calc(100% - 2px);
  border-color: ${(props) => {
    if (props.always && props.light) {
      return "white";
    } else if (props.always) {
      return silver_polish;
    } else {
      return "transparent";
    }
  }};
  border-width: ${(props) => (props.thin ? 0.5 : 1)}px;
  border-block-start: none;
  margin: 0px;

  @media (min-width: ${desktopBreakpoint}px) {
    border-color: ${(props) => (props.light ? "white" : silver_polish)};
    border-width: ${(props) => (props.thin ? 0.5 : 1)}px;
    position: absolute;
    left: 0px;
    margin: 0px;
  }
`;
