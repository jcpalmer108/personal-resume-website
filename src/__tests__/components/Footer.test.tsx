import { render, screen } from "@testing-library/react";
import Footer from "../../components/Footer";
import { SeparatingLineProps } from "../../types/SeparatingLine";

// mock components
const mockSeparatingLine = jest.fn();
jest.mock(
  "../../components/SeparatingLine",
  () =>
    ({ light, thin, always }: SeparatingLineProps) => {
      mockSeparatingLine(light, thin, always);
      return <hr />;
    }
);

const mockSection = jest.fn();
jest.mock(
  "../../components/Section",
  () =>
    ({
      label,
      children,
      light,
      thin,
      wide,
      center,
      bottom,
      noTop,
      noBorder,
    }: SectionProps) => {
      mockSection(label, light, thin, wide, center, bottom, noTop, noBorder);
      return <div>{children}</div>;
    }
);

// params needed for component
const params = {
  required: {
    menu: [
      {
        key: "one",
        label: "One",
      },
      {
        key: "two",
        label: "Two",
      },
    ],
  },
};

describe("Footer", () => {
  test("renders if only required params are passed in", () => {
    // given
    const { required } = params;
    render(<Footer menu={required.menu} />);

    // then
    expect(screen.getByTestId("Footer")).toMatchSnapshot();
    expect(screen.getByTestId("Copyright")).toBeTruthy();

    expect(screen.getByTestId("Footer Menu").childNodes).toHaveLength(
      required.menu.length
    );
    required.menu.forEach((link, index) => {
      expect(
        screen.getByTestId("Footer Menu").childNodes[index]
      ).toHaveTextContent(link.label.toUpperCase());
    });

    expect(mockSeparatingLine).toHaveBeenCalled();
    expect(mockSeparatingLine).toHaveBeenCalledWith(undefined, true, true);

    expect(mockSection).toHaveBeenCalledTimes(2);
    expect(mockSection).toHaveBeenNthCalledWith(
      1,
      undefined,
      undefined,
      true,
      true,
      undefined,
      undefined,
      undefined,
      undefined
    );
    expect(mockSection).toHaveBeenNthCalledWith(
      2,
      undefined,
      undefined,
      true,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined
    );
  });
});
