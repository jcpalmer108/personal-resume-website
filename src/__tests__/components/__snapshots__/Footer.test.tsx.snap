// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders if only required params are passed in 1`] = `
<div
  class="sc-bczRLJ ggSMEg"
  data-testid="Footer"
>
  <img
    class="sc-gsnTZi cWKKaQ"
    src="jennapalmer-light.svg"
  />
  <div
    class="sc-dkzDqf diyymw"
    data-testid="Footer Menu"
  >
    <a
      class="sc-hKMtZM hNKKYy"
      href="#one"
    >
      ONE
    </a>
    <a
      class="sc-hKMtZM hNKKYy"
      href="#two"
    >
      TWO
    </a>
  </div>
</div>
`;
